src/compiler/crystal/tools/doc/macro.cr

```diff
@@ -29,8 +29,7 @@ class Crystal::Doc::Macro
 
   def id
     String.build do |io|
-      io << to_s.gsub(/<.+?>/, "").gsub(' ', "")
-      io << "-macro"
+      io << to_s.gsub(/<.+?>/, "").delete(' ') << "-macro"
```

Comment:
Pulling `<< "-macro"` into the previous line is an unrelated change. It doesn't really improve performance like the rest.
And I find it is easier to read on a separate line.